### PR TITLE
Use eslint and @kentcdodds' presets

### PR DESCRIPTION
**Pull request by gtramontina**
This introduces a "breaking change". Because we've adopted ES2015, node >= `4.0` is required.

### CODE REVIEW

**Comment by kentcdodds** [comment]
Neat!!! :D

**Comment by gtramontina** [comment]
Let me know if there are any outstanding issues preventing this from getting merged! :wink: :beers:

**Comment by kentcdodds** [comment]
Hey @gtramontina, thanks for being willing to collaborate on your own project ;-)

So I have one problem with this and that is that it drops support for Node 0.10.x and 0.12.x. This may not seem like a big deal, but the vast majority of developers are still on 0.12.x unfortunately.

Here's what I'd recommend, check out my series on creating an open source library (free on egghead) [here](http://kcd.im/write-oss). Specifically look for the ES6 stuff. One thing to note is that it's using an old version of Babel. I recommend you try to get the latest version of Babel working instead. See [this branch](https://github.com/kentcdodds/starwars-names/tree/wip/mocha-to-ava) to see how that's set up (still need to update the lesson, also don't be concerned with migrating our tests to ava as I do in that branch).

I definitely think it's important that we support older versions of Node.

**Comment by gtramontina** [comment]
Oh yeah! Totally! How could I forget about babel? Thanks for reminding me of it.
I'll do it as soon as I can. Cheers! :beers:

**Comment by gtramontina** [comment]
It seems that there's something funky going on with `mock-fs` after transpiling with `babel`. One of the tests break because it is reaching the actual filesystem... I pushed the changes up to get some help. :smile:

ps.: `prepublish` script is not there yet.
ps2.: commit message needs work. squashing/amending needed.

**Comment by kentcdodds** [comment]
Hmmm.... I'm not sure. I'm pretty sure that babel's trying to get the config from the `package.json`. I think we need to mock the filesystem in a way that doesn't prevent babel from doing that...

**Comment by ta2edchimp** [comment]
The broken test can be fixed by [using `proxyquire` to simulate the absence of the `ghooks` module](https://github.com/ta2edchimp/ghooks/blob/gtramontina-eslint/test/hook.template.raw.test.js#L19-L26) directly, without further requirement to fiddle around with / mock anything on the fs.

I made the dedicated Pull Request #52 to branch `master` to show that the particular test now passes. (In return, there's still something different broken for Node versions 0.10 and 0.12 ☹️)

Please note: I had to add `babel-cli` as a devDependency and alter the `npm script` [`coverage`](https://github.com/ta2edchimp/ghooks/blob/gtramontina-eslint/package.json#L25) to run `istanbul` using `babel-node`, due to [a known bug](https://github.com/gotwarlost/istanbul/issues/496) I ran into.

**Comment by gtramontina** [comment]
Thank you very much @ta2edchimp! I'll squash/amend my commits and merge this PR as soon as I get my hands on my laptop... There's only so much one can do from their mobile... :-D

**Comment by ta2edchimp** [comment]
Cool. I'll get my hands on #50 in the meanwhile ...

**Comment by gtramontina** [comment]
Travis looks happy, btw... :+1:

**Comment by gtramontina** [comment]
@ta2edchimp I've just added you as a collaborator so you and @kentcdodds can make decisions as well! I'm not always available!

Thanks again for your contributions!

**Comment by ta2edchimp** [comment]
I just noticed, we could also get rid of the `babel-cli` dependency (introduced with ea5b22c, but obsolete as of cb5187a), but now *I* am off my laptop 😖

**Comment by gtramontina** [comment]
So… I've squashed/amended some of the commits. Tried to make justice and keep @ta2edchimp as the author of the bulk of work. :smile: 

I ended up having to bring `babel-cli` back in order to build `dist` (e4be646)… `¯\_(ツ)_/¯`

_But_ now I hit a road block… We currently have an install script that runs every time someones installs `ghooks`, which (now) executes `dist/install`. The problem happens when someone checks-out this project and runs `npm install`… the same script kicks in and breaks because there's no `dist/` built. This is the reason Travis is broken.

Here's a couple ways I can think of about tackling this:

1. Find a way to no run `scripts.install` when running `npm install` on the project itself, only when being installed as a module.
2. `cp lib/* dist/` on `scripts.preinstall`

I dislike them both, lol… So any suggestions are more than welcome!

**Comment by ta2edchimp** [comment]
I think this is one of the cases where `build ≠ dist`, as this project's build process depends on its very own functionality.

As I see it, it is correct to **not** mention `dist` in `.npmignore`, but I perceive its mention in `.gitignore` as a _misconception_: Rather should a directory named `build` be `.npmignored` and `.gitignored`. And the build process's results (output to `build`) should replace `dist`'s contents only _after_ a successful build, thus providing a cloned repo with a working copy of its own module.

A folder `dist` is always included (repo & npm), always providing the latest successful build.

A folder `build` is created during the build process and its contents replace those of `dist` only after a successful build.
This folder is `.npmignore`d and `.gitignore`d.

**So basically:**

- replace `dist` by `build` in `.gitignore`,
- add `build` to `.npmignore`,
- change `npm scripts`:

>```JSON
>  "scripts": {
>    "prebuild": "rm -rf build && mkdir build",
>    "build": "babel lib --out-dir build && babel lib/hook.template.raw --out-file build/hook.template.raw",
>    "postbuild": "rm -rf dist && mkdir dist && cp build/* dist"
>  }
>```

This way one could successfully clone / fetch a repo with a broken build and work on it, without lacking ghooks' functionality.

... or am I missing something?
